src/aurora/textureatlasfile.cpp

```diff
@@ -0,0 +1,109 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file
+ * A loader class for texture atlas xml files for dragonage and dragonage2
+ */
+
+#include "src/common/ustring.h"
+#include "src/common/readstream.h"
+#include "src/common/xml.h"
+#include "src/common/streamtokenizer.h"
+#include "src/common/strutil.h"
+
+#include "src/aurora/resman.h"
+#include "src/aurora/textureatlasfile.h"
+
+namespace Aurora {
+
+TextureAtlasFile::TextureAtlasFile(Common::UString name) {
+	Common::ScopedPtr<Common::SeekableReadStream> stream(ResMan.getResource(name, kFileTypeXML));
+	load(*stream);
+}
+
+TextureAtlasFile::TextureAtlasFile(Common::SeekableReadStream &stream) {
+	load(stream);
+}
+
+bool TextureAtlasFile::isAtlasTexture(const Common::UString &texture) const {
+	return _atlasTextures.find(texture) != _atlasTextures.end();
+}
+
+void TextureAtlasFile::getAtlasTexture(const Common::UString &texture,
+									   Common::UString &textureFile, float &x, float &y, float &w, float &h) const {
+	std::map<Common::UString, AtlasTexture>::const_iterator iter = _atlasTextures.find(texture);
+	if (iter == _atlasTextures.end()) {
+		throw Common::Exception("Atlas Texture %s not found", texture);
+	}
+
+	AtlasTexture atlasTexture = iter->second;
```

Comment:
`const AtlasTexture &atlasTexture`. You don't need to copy the whole structure again.